Sources/StreamChat/Models/ChatMessageImageAttachment.swift

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A type representing an attachment of type `image`.
+public struct ChatMessageImageAttachment: ChatMessageAttachment, AttachmentEnvelope, Decodable {
+    public var type: AttachmentType { .image }
+    /// A unique identifier of the attachment.
+    public var id: AttachmentId?
+    
+    /// When a new attachment is created, this value contains the URL of the source from which the attachment
+    /// data are uploaded to the server. For already sent attachments this value is usually `nil`. This value is
+    /// device-specific and is not synced with other devices.
+    public var localURL: URL?
+    /// A local attachment state
+    public var localState: LocalAttachmentState?
+    
+    /// A title, usually the name of the image.
+    public let title: String?
+    /// A link to the image.
+    public var imageURL: URL?
+    /// A link to the image preview.
+    public let imagePreviewURL: URL?
+    
+    private enum CodingKeys: String, CodingKey {
+        case title
+        case type
+        case image
+        case url
+        case name
+        case thumbURL = "thumb_url"
+        case fallback
+        case imageURL = "image_url"
+        case assetURL = "asset_url"
+    }
```

Comment:
Maybe we should have a common `CodingKeys` for all built-in types? Like we have it for user payloads, message payloads, etc.

Reply:
Nice catch, thanks!